### PR TITLE
feat: add testmon in pytest

### PR DESCRIPTION
**Pull request by jiashenC**
Add `testmon` in pytest

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -251,6 +267,20 @@ jobs:
...
+    # Collect the testmondata only on the staging branch 
+      - when:
+          condition:
+            and:
+              - equal: [ staging, << pipeline.git.branch >> ]
+              - equal: [ LONG INTEGRATION, << parameters.mode >> ]
+              - equal: [ "3.10", << parameters.v >> ]
+              - equal: [ DISABLED, << parameters.ray >> ] 
+          steps:
+          - save_cache:
```
Comment (xzdandy) [inline .circleci/config.yml:279]: Not clear why we don't need a indentation here. But circle ci complains when we have an indentation here.

**Comment by jiashenC** [comment]
> Do we also need to push the `.testmondata` file?

It will be generated from the `staging` branch and used for open PRs.

File: .circleci/config.yml [hunk 2]
```diff
@@ -201,6 +212,11 @@ jobs:
       - restore_cache:
           keys:
             - v1-model_cache-{{ checksum "setup.py" }}
+
+      # Restore testmondata
+      - restore_cache:
```
Comment (jiashenC) [inline .circleci/config.yml:217]: Not something critical, shall we only restore the cache for the open PR long integration cached test?

Comment (xzdandy) [reply .circleci/config.yml:217]: Actually I tried a different way to simplify the circle ci config. Please take a look.

**Comment by xzdandy** [comment]
@jiashenC @gaurav274 This PR is ready. Please review. Thanks.

**Review by xzdandy — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (jiashenC) [reply .circleci/config.yml:217]: Oh, I meant it is currently restoring the cache for any test. Is that expected? Do we want to rerun everything for `staging` and `master`? Or we should run only affected as well?

**Pull request merged**